.travis.yml

```diff
@@ -81,19 +68,14 @@ install:
   - apt-cache policy docker-ce
   - sudo apt-get install -y docker-ce
   - sudo usermod -aG docker ${USER}
-#  - if [ "$ANSIBLE_VERSION" = "latest" ]; then pip install ansible[azure]; else pip
-#    install ansible[azure]==$ANSIBLE_VERSION; fi
-#  - pip install ansible[azure]==2.6.0.rc3
-#  - pip install ansible[azure]==2.5.0
-  - git clone https://github.com/ansible/ansible.git
-  - pip install -r ./ansible/requirements.txt
-  - pip install -r ./ansible/packaging/requirements/requirements-azure.txt 
-  - cd ansible
-  - source ./hacking/env-setup
-  - cd ..
+  - sudo apt-get install -y default-jdk
+  - sudo apt-get install -y maven
+
+  - if [ "$ANSIBLE_VERSION" = "latest" ]; then pip install ansible[azure]; else pip
+    install ansible[azure]==$ANSIBLE_VERSION; fi
   - ansible --version
-#  - ansible-galaxy install Azure.azure_preview_modules
-#  - pip install -I -r ~/.ansible/roles/Azure.azure_preview_modules/files/requirements-azure.txt
+  - ansible-galaxy install Azure.azure_preview_modules
+  - pip install -I -r ~/.ansible/roles/Azure.azure_preview_modules/files/requirements-azure.txt
```

Comment:
if this is done, we are not testing against latest version of Ansible, but against azure_preview_modules. so some test may get broken, some other tests may work correctly even if they wouldn't work on bare Ansible setup

Reply:
I've tested the repo, no failed test now. If a sample playbook could run passed against role, then it's ok for user. the dependency issue you're worried about didn't happen. https://travis-ci.org/Azure-Samples/ansible-playbooks/builds/400277345

Reply:
but you are not really testing against 2.5, you are testing against the role.

Reply:
My understanding is if the playbook is supported by Ansible released version properly, we just test this with Ansible bare install. In the playbook, there is no `role` field.

However, some feature is not included in the Ansible released version, user need to install role and add a `role` field in the playbook to define these tasks should be run with role. For these playbook, failed on bare install is as expected.

Reply:
well, the dependency didn't happen, because the test is running against Ansible 2.6 right, now, and apparently all the fixes are now released.
i have updated all the failing modules and all the dependencies in 2.6, so you can't see dependencies issue.
But if you try to run against Ansible 2.5 (which was still a case maybe a week ago), the test would fail.
And it may start failing if any dependencies change in develop branch, and will be propagated to azure_preview_modules.
So I still think we should have separate test environment for azure_preview_modules.

Reply:
agree with Yuwei. what we did in travis is just mimic how user will use ansible and role. If some bugfixing/features in role, user's environment is just same as this travis environment. didn't see any risk of breaking thing.

Reply:
@zikalino  could you pls share what the depenedency issue you mentioned? rdbms? could you pls share the link? If some playbook should be passed against 2.5, but failed now, it seems bug in 2.5. Then let's fix the bug in ansible core, we'll need add test in ansible core to cover the missed p art. If our solution to user is we're not fixing it in 2.5, but fixing it in role, then current sample repo ci is working exactly same as users'.

Reply:
no, it's not a bug, if you use Ansible 2.5 it may have older modules, and older dependencies. Installing preview modules, will updated dependencies, and versions may be too high